segment.go

```diff
@@ -259,27 +265,30 @@ func (seg *segment) writeInternal(data []byte, chunkType ChunkType) error {
 	binary.LittleEndian.PutUint32(seg.header[:4], sum)
 
 	// append the header and data to segment file
-	buf := bytebufferpool.Get()
-	defer func() {
-		bytebufferpool.Put(buf)
-	}()
-	buf.B = append(buf.B, seg.header...)
-	buf.B = append(buf.B, data...)
-	if _, err := seg.fd.Write(buf.Bytes()); err != nil {
-		return err
-	}
+	seg.chunkBuffer = append(seg.chunkBuffer, seg.header...)
+	seg.chunkBuffer = append(seg.chunkBuffer, data...)
+}
 
+func (seg *segment) flushChunkBuffer() error {
 	if seg.currentBlockSize > blockSize {
 		panic("wrong! can not exceed the block size")
 	}
 
+	if _, err := seg.fd.Write(seg.chunkBuffer); err != nil {
+		return err
+	}
+
 	// update the corresponding fields
-	seg.currentBlockSize += dataSize + chunkHeaderSize
-	// A new block
-	if seg.currentBlockSize == blockSize {
-		seg.currentBlockNumber += 1
-		seg.currentBlockSize = 0
+	seg.currentBlockSize += uint32(len(seg.chunkBuffer))
+
+	if seg.currentBlockSize >= blockSize {
```

Comment:
Will seg.currentBlockSize be greater than blockSize?
I think they are equal at most.

Reply:
@roseduan it can be greater than blockSize as I add the chunkBuffer size to it at line 282